workspaces/adventure-pack/goodies/java/src/digits_int/Test.java

```diff
@@ -0,0 +1,72 @@
+package digits_int;
+
+import static org.junit.jupiter.api.Assertions.assertIterableEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import iterable_int_stream.IterableIntStream;
+import java.util.stream.IntStream;
+import org.junit.jupiter.api.Test;
+
+class DigitsIntTest {
+
+  @Test
+  public void withIntegerGreaterThanZero() {
+    try {
+      var digits = AP.digits(4984);
+      assertIterableEquals(
+        IterableIntStream.from(IntStream.of(4, 8, 9, 4)),
+        digits,
+        "Test Failed. Digits generated for positive integer are not correct."
+      );
+    } catch (Exception e) {
+      e.printStackTrace();
+      fail();
+    }
+  }
+
+  @Test
+  public void withZero() {
+    try {
+      var digits = AP.digits(0);
+      assertIterableEquals(
+        IterableIntStream.from(IntStream.of(0)),
+        digits,
+        "Test Failed. Digits generated for integer 0 are not correct."
+      );
+    } catch (Exception e) {
+      e.printStackTrace();
+      fail();
+    }
+  }
+
+  @Test
+  public void withRadixOtherThan10() {
+    try {
+      var digits = AP.digits(12345, 7);
+      assertIterableEquals(
+        IterableIntStream.from(IntStream.of(4, 6, 6, 0, 5)),
+        digits,
+        "Test Failed. Digits generated for the integer with radix other than 10 are not correct."
```

Comment:
Can we make the test fail and see what error message we see? My guess is that we will get something very similar to this already. If so, we can omit the message if it's not adding new information.

(If it is adding new information let's keep it of course.)

Reply:
hey, the error message is not exactly same but I guess it it enough to debug. So, I think we can omit the message.
 
![utest](https://github.com/user-attachments/assets/e7af5266-99f6-4caa-9d48-2fc147592638)

Reply:
I think our own custom message can be useful, if we add more information that is not otherwise available in the error message. Also fine to omit in this case.

Reply:
Addressed.